Failed: ClientServerSpec.The low-level HTTP infrastructure should produce a useful error message when connecting to a HTTP endpoint over HTTPS

```
java.lang.AssertionError: timeout (3 seconds) waiting for 6 messages on WarningFilter(None,Right(Perhaps this was an HTTPS request sent to an HTTP endpoint),false)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:114)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16$$anonfun$apply$mcV$sp$17.apply$mcV$sp(ClientServerSpec.scala:719)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16$$anonfun$apply$mcV$sp$17.apply(ClientServerSpec.scala:713)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16$$anonfun$apply$mcV$sp$17.apply(ClientServerSpec.scala:713)
	at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:34)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16.apply$mcV$sp(ClientServerSpec.scala:713)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16.apply(ClientServerSpec.scala:713)
	at akka.http.scaladsl.ClientServerSpec$$anonfun$1$$anonfun$apply$mcV$sp$16.apply(ClientServerSpec.scala:713)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
...
```

https://jenkins.akka.io:8498/job/akka-http-nightly-against-akka-master-on-2.11/458/testReport/junit/akka.http.scaladsl/ClientServerSpec/The_low_level_HTTP_infrastructure_should_produce_a_useful_error_message_when_connecting_to_a_HTTP_endpoint_over_HTTPS/

--- a/akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala
+++ b/akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala
@@ -92,17 +92,11 @@ private[http] final class HttpRequestParser(
               }
             case c ⇒ parseCustomMethod(ix + 1, sb.append(c))
           }
-        } else {
-          if (sb.length > 0 && sb.charAt(0) == 0x16)
-            throw new ParsingException(
-              BadRequest,
-              ErrorInfo("Unsupported HTTP method", s"The HTTP method started with 0x16 rather than any known HTTP method. " +
-                "Perhaps this was an HTTPS request sent to an HTTP endpoint?"))
-          else throw new ParsingException(
+        } else
+          throw new ParsingException(
             BadRequest,
             ErrorInfo("Unsupported HTTP method", s"HTTP method too long (started with '${sb.toString}'). " +
               "Increase `akka.http.server.parsing.max-method-length` to support HTTP methods with more characters."))
-        }
 
       @tailrec def parseMethod(meth: HttpMethod, ix: Int = 1): Int =
         if (ix == meth.value.length)
@@ -127,7 +121,12 @@ private[http] final class HttpRequestParser(
         case 'O' ⇒ parseMethod(OPTIONS)
         case 'T' ⇒ parseMethod(TRACE)
         case 'C' ⇒ parseMethod(CONNECT)
-        case _   ⇒ parseCustomMethod()
+        case 0x16 ⇒
+          throw new ParsingException(
+            BadRequest,
+            ErrorInfo("Unsupported HTTP method", s"The HTTP method started with 0x16 rather than any known HTTP method. " +
+              "Perhaps this was an HTTPS request sent to an HTTP endpoint?"))
+        case _ ⇒ parseCustomMethod()
       }
     }
 

